src/Game.java

```diff
@@ -1,45 +1,119 @@
+        if (!allPlayers.stream().map(Object::toString)
+            .collect(Collectors.toList()).contains(playerName)) {
+          if (playerName.length() != 0) {
+            break;
+          }
+          System.out.println("Invalid name, please enter another name");
+        } else {
+          System.out.println("Please enter another name, " + playerName
+              + " has already been taken:");
         }
+      }
 
-        int currentLeaderIndex = 0;
+      allPlayers.add(new Player(playerName));
+    }
+
+    int currentLeaderIndex = 0;
+
+    mainLoop:
+    while (true) {
 
-        while (true) {
-            System.out.println("Do you wish to start a new game? Y/N");
-            String answer = scanner.nextLine();
+      Player leader = allPlayers.get(currentLeaderIndex);
 
-            if (answer.startsWith("N")) {
-                break;
-            }
+      Word word;
+      System.out.println(leader + ", you are the leader, please pick a word:");
 
-            Player leader = allPlayers.get(currentLeaderIndex);
+      while (true) {
+        word = new Word(scanner.nextLine());
+        if (word.wordLength() != 0) {
+          break;
+        }
+        System.out.println("Please enter a word with at least 1 letter:");
+      }
+
+      List<Player> players = new ArrayList<>(allPlayers);
+      players.remove(leader);
 
-            System.out.println(leader + ", you are the leader, please pick a word:");
+      Session session = new Session(scanner, players, leader, word);
+      session.play();
 
-            Word word = new Word(scanner.nextLine());
+      currentLeaderIndex = (currentLeaderIndex + 1) % allPlayers.size();
+      System.out.println("Current Standings: ");
+      System.out.println(leaderboardToSet(allPlayers));
 
-            List<Player> players = new ArrayList<>(allPlayers);
-            players.remove(leader);
+      System.out.println("Do you wish to start a new game? Y/N");
+      String answer = scanner.nextLine();
 
-            Session session = new Session(scanner, players, leader, word);
-            session.play();
+      while (!answer.startsWith("y") && !answer.startsWith("Y")) {
+        if (answer.startsWith("N") || answer.startsWith("n")) {
+          break mainLoop;
+        } else {
+          System.out.println("Sorry, couldn't understand");
+          System.out.println("Do you wish to start a new game? Y/N");
+          answer = scanner.nextLine();
+        }
+      }
 
-            currentLeaderIndex = (currentLeaderIndex + 1) % allPlayers.size();
+      System.out.println("Thank you for playing! These are our top players:");
+      int topScore = allPlayers.get(0).getScore();
+      List<Player> topPlayers = new ArrayList<>();
+      for (Player p : allPlayers) {
+        if (p.getScore() >= topScore) {
+          topPlayers.add(p);
+          topScore = p.getScore();
         }
+      }
+      topPlayers.forEach(p -> System.out.print(p + ", "));
```

Comment:
Ah, so this makes me think that you just wanted to print the players which all shared the same top score. Well, the above loop needs some modification for that to happen. Also, I would move this whole chunk of logic into a function like `displayLeaderboard` or something.

You should aim to split up massive function like this entire `main()` into smaller ones, appropriately named. Then your code becomes self-documenting, as the way you name the functions should concisely express your intent at every stage of this main function.